lib/GridItem.jsx

```diff
@@ -415,6 +458,106 @@ export default class GridItem extends React.Component<Props, State> {
     );
   }
 
+  delayEvents = React.createRef();
+  addEvent = (type, event, passive = true) => {
+    if (this.elementRef.current) {
+      this.elementRef.current.addEventListener(type, event, {
+        passive
+      });
+      this.delayEvents.current.push({ type, event, passive });
+    }
+  };
+
+  removeEvents = () => {
+    if (this.elementRef.current) {
+      this.delayEvents.current.forEach(({ type, event, passive }) => {
+        this.elementRef.current.removeEventListener(type, event, {
+          passive
+        });
+      });
+      this.delayEvents.current = [];
+    }
+  };
+
+  //A reference to the timeout handler to be able to access it at any time.
+  timeoutRef: { current: null | TimeoutID, ... } = React.createRef();
+
+  /**
+   * Start the delayed counter to determine when a drag should start.
+   * @param  {Event} e          TouchStart/MouseDown event.
+   */
+  startDelayTimeout: Event => void = e => {
+    this.delayEvents.current = [];
+    if (!this.state.allowedToDrag) {
+      /**
+       * Register events to cancel the timeout handler if user releases the mouse or touch
+       */
+      this.addEvent("mouseup", this.resetDelayTimeout);
+      this.addEvent("touchend", this.resetDelayTimeout);
+      /**
+       * Prevent user from doing touch and scroll at the same time.
+       * If the user starts scrolling, we can not cancel the scroll event,
+       * so we cancel the drag event instead.
+       */
+      this.addEvent("touchmove", this.handleTouchMove, false);
```

Comment:
Why doesn't `this.handleTouchMove()` serve this purpose?

Reply:
This part is the most important part of delayed drag feature and cannot be removed.

We must register a non-passive touchmove event to be able to prevent scrolls when the drag delay timer has ended.

Another important note is that we must simply cancel the whole delayed drag event and reset everything if the user moves his finger before the timer is up. The reason for this is that if a user starts scrolling even for one pixel, we cannot stop the browser from scrolling afterward and e.preventDefault will not be effective.